File: CS/WebSite/App_Code/UserAppointmentFormClass.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.ASPxScheduler.Internal;
using DevExpress.XtraScheduler;
using DevExpress.Web.ASPxEditors;

#region #userappointmentformtemplatecontainer
public class UserAppointmentFormTemplateContainer : AppointmentFormTemplateContainer {
    public UserAppointmentFormTemplateContainer(ASPxScheduler control)
        : base(control) {
    }
        public double Field1 {
        get {
            object val = Appointment.CustomFields["Field1"];
            return (val == DBNull.Value) ? 0 : Convert.ToDouble(val);
        }
    }
}
#endregion #userappointmentformtemplatecontainer
#region #userappointmentformcontroller
public class UserAppointmentFormController : AppointmentFormController {
    public UserAppointmentFormController(ASPxScheduler control, Appointment apt)
        : base(control, apt) {
    }
    public double Field1 { 
        get { return (double)EditedAppointmentCopy.CustomFields["Field1"]; } 
        set { EditedAppointmentCopy.CustomFields["Field1"] = value; } }

    double SourceField1 { 
        get { return (double)SourceAppointment.CustomFields["Field1"]; } 
        set { SourceAppointment.CustomFields["Field1"] = value; } }

    public override bool IsAppointmentChanged() {
        if(base.IsAppointmentChanged())
            return true;
        return SourceField1 != Field1;
    }
    protected override void ApplyCustomFieldsValues() {
        SourceField1 = Field1;
    }
}
#endregion #userappointmentformcontroller
#region #userappointmentsavecallbackcommand
public class UserAppointmentSaveCallbackCommand : AppointmentFormSaveCallbackCommand {
    public UserAppointmentSaveCallbackCommand(ASPxScheduler control)
        : base(control) {
    }
    protected internal new UserAppointmentFormController Controller { 
        get { return (UserAppointmentFormController)base.Controller; } }

    protected override void AssignControllerValues() {
        ASPxTextBox tbPrice = (ASPxTextBox)FindControlByID("tbPrice");

        Controller.Field1 = Convert.ToDouble(tbPrice.Text);

        base.AssignControllerValues();        
    }
    protected override AppointmentFormController CreateAppointmentFormController(Appointment apt) {
        return new UserAppointmentFormController(Control, apt);
    }
}
#endregion #userappointmentsavecallbackcommand

File: VB/WebSite/App_Code/UserAppointmentFormClass.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports DevExpress.Web.ASPxScheduler
Imports DevExpress.Web.ASPxScheduler.Internal
Imports DevExpress.XtraScheduler
Imports DevExpress.Web.ASPxEditors

#Region "#userappointmentformtemplatecontainer"
Public Class UserAppointmentFormTemplateContainer
	Inherits AppointmentFormTemplateContainer
	Public Sub New(ByVal control As ASPxScheduler)
		MyBase.New(control)
	End Sub
		Public ReadOnly Property Field1() As Double
		Get
			Dim val As Object = Appointment.CustomFields("Field1")
			If (val Is DBNull.Value) Then
				Return 0
			Else
				Return Convert.ToDouble(val)
			End If
		End Get
		End Property
End Class
#End Region ' #userappointmentformtemplatecontainer
#Region "#userappointmentformcontroller"
Public Class UserAppointmentFormController
	Inherits AppointmentFormController
	Public Sub New(ByVal control As ASPxScheduler, ByVal apt As Appointment)
		MyBase.New(control, apt)
	End Sub
	Public Property Field1() As Double
		Get
			Return CDbl(EditedAppointmentCopy.CustomFields("Field1"))
		End Get
		Set(ByVal value As Double)
			EditedAppointmentCopy.CustomFields("Field1") = value
		End Set
	End Property

	Private Property SourceField1() As Double
		Get
			Return CDbl(SourceAppointment.CustomFields("Field1"))
		End Get
		Set(ByVal value As Double)
			SourceAppointment.CustomFields("Field1") = value
		End Set
	End Property

	Public Overrides Function IsAppointmentChanged() As Boolean
		If MyBase.IsAppointmentChanged() Then
			Return True
		End If
		Return SourceField1 <> Field1
	End Function
	Protected Overrides Sub ApplyCustomFieldsValues()
		SourceField1 = Field1
	End Sub
End Class
#End Region ' #userappointmentformcontroller
#Region "#userappointmentsavecallbackcommand"
Public Class UserAppointmentSaveCallbackCommand
	Inherits AppointmentFormSaveCallbackCommand
	Public Sub New(ByVal control As ASPxScheduler)
		MyBase.New(control)
	End Sub
	Protected Friend Shadows ReadOnly Property Controller() As UserAppointmentFormController
		Get
			Return CType(MyBase.Controller, UserAppointmentFormController)
		End Get
	End Property

	Protected Overrides Sub AssignControllerValues()
		Dim tbPrice As ASPxTextBox = CType(FindControlByID("tbPrice"), ASPxTextBox)

		Controller.Field1 = Convert.ToDouble(tbPrice.Text)

		MyBase.AssignControllerValues()
	End Sub
	Protected Overrides Function CreateAppointmentFormController(ByVal apt As Appointment) As AppointmentFormController
		Return New UserAppointmentFormController(Control, apt)
	End Function
End Class
#End Region ' #userappointmentsavecallbackcommand